data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriter.java

```diff
@@ -434,175 +144,12 @@ public void addValue(int rowId, BigDecimal data, ColumnVector output) {
-    }
-
-    @Override
-    public Class<Map> getJavaClass() {
-      return Map.class;
-    }
-
-    @Override
-    @SuppressWarnings("unchecked")
-    public void addValue(int rowId, Map data, ColumnVector output) {
-      if (data == null) {
-        output.noNulls = false;
-        output.isNull[rowId] = true;
-      } else {
-        output.isNull[rowId] = false;
-        Map<Object, Object> map = (Map<Object, Object>) data;
-        List<Object> keys = Lists.newArrayListWithExpectedSize(map.size());
-        List<Object> values = Lists.newArrayListWithExpectedSize(map.size());
-        for (Map.Entry<?, ?> entry : map.entrySet()) {
-          keys.add(entry.getKey());
-          values.add(entry.getValue());
-        }
-        MapColumnVector cv = (MapColumnVector) output;
-        // record the length and start of the list elements
-        cv.lengths[rowId] = map.size();
-        cv.offsets[rowId] = cv.childCount;
-        cv.childCount += cv.lengths[rowId];
-        // make sure the child is big enough
-        cv.keys.ensureSize(cv.childCount, true);
-        cv.values.ensureSize(cv.childCount, true);
-        // Add each element
-        for (int e = 0; e < cv.lengths[rowId]; ++e) {
-          int pos = (int) (e + cv.offsets[rowId]);
-          keyConverter.addValue(pos, keys.get(e), cv.keys);
-          valueConverter.addValue(pos, values.get(e), cv.values);
-        }
-      }
-    }
-  }
-
-  private static Converter buildConverter(TypeDescription schema) {
-    switch (schema.getCategory()) {
-      case BOOLEAN:
-        return new BooleanConverter();
-      case BYTE:
-        return new ByteConverter();
-      case SHORT:
-        return new ShortConverter();
-      case DATE:
-        return new DateConverter();
-      case INT:
-        return new IntConverter();
-      case LONG:
-        String longAttributeValue = schema.getAttributeValue(ORCSchemaUtil.ICEBERG_LONG_TYPE_ATTRIBUTE);
-        ORCSchemaUtil.LongType longType = longAttributeValue == null ? ORCSchemaUtil.LongType.LONG :
-            ORCSchemaUtil.LongType.valueOf(longAttributeValue);
-        switch (longType) {
-          case TIME:
-            return new TimeConverter();
-          case LONG:
-            return new LongConverter();
-          default:
-            throw new IllegalStateException("Unhandled Long type found in ORC type attribute: " + longType);
-        }
-      case FLOAT:
-        return new FloatConverter();
-      case DOUBLE:
-        return new DoubleConverter();
-      case BINARY:
-        String binaryAttributeValue = schema.getAttributeValue(ORCSchemaUtil.ICEBERG_BINARY_TYPE_ATTRIBUTE);
-        ORCSchemaUtil.BinaryType binaryType = binaryAttributeValue == null ? ORCSchemaUtil.BinaryType.BINARY :
-            ORCSchemaUtil.BinaryType.valueOf(binaryAttributeValue);
-        switch (binaryType) {
-          case UUID:
-            return new UUIDConverter();
-          case FIXED:
-            return new FixedConverter();
```

Comment:
Is there problem here ? I mean the iceberg  FIXED type is mapping to `ByteBuffer` class and the `FixedConverter`  will write a `byte[]` instance to the `ColumnVector`,  I guess the unit tests did not expose this issue because the `RandomDataGenerator` are producing a `ByteBuffer` instance and we did not use the TypeID#javaClass(). The `BINARY` & `TIME` have similar issues I think.    @rdsr @shardulm94  FYI.

Reply:
`TypeID#javaClass()` does not define the Java types exposed by the `iceberg-data` module. They are mainly used internally in `iceberg-core` when serializing and deserializing to/from manifest files and in Expressions as literals. Iceberg Generics use `byte[]` for fixed types. So this looks correct and is in parity with Parquet and Avro Generics. You can also look at Timestamp types. Iceberg Generics uses `LocalDateTime` and `OffsetDateTime` for Timestamp Without Zone and Timestamp With Zone respectively. However both types map to longs in `TypeID#javaClass()`.

Reply:
I see.  The TypeID#javaClass defines the class of iceberg desired data types(expressions as you said), the `Record`'s data type is different, we also have a class to transform between them . https://github.com/apache/iceberg/blob/68e417c3591439044b229b657a529cad0fec66e3/data/src/main/java/org/apache/iceberg/data/InternalRecordWrapper.java#L45 
But another question is: why we need to use the different data types in `Expression` or manifest files serialization & deserialization, which produces this complexity....

Reply:
Seems the `TypeID#javaClass`  defines the literal type, means the type to compare or serialize/deserialize. Actually, I'd prefer to  pass the `LocalDateTime` object to the comparator and do the `LocalDateTime` to `Long` conversion when comparing.  Then  we hidden the literal types between different types, and the upper layer won't to  wrap again and again.

Reply:
> I'd prefer to pass the LocalDateTime object to the comparator and do the LocalDateTime to Long conversion when comparing

Iceberg's internal representation does not use higher-level types like LocalDateTime for a few good reasons:
1. It is simpler to work with ordinal values
2. The interpretation of an ordinal value is delegated to the object model: Iceberg is agnostic to calendars, time zones, and other concerns that are built into the processing engines
3. The guarantee is simpler: whatever data values are passed into Iceberg will be passed back out, unmodified